File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
from collections import deque

def bfs(start_node, graph):
    queue = deque([start_node])
    visited = {start_node}

    while queue:
        curr_node = queue.popleft()

        for next_node in graph[curr_node]:
            if next_node not in visited:
                visited.add(next_node)
                queue.append(next_node)
    return visited

# 딕셔너리 그래프
n = int(input())
dic = {}
m = int(input())

for i in range(m):
    key, value = map(int, sys.stdin.readline().split())
    if key not in dic:
        dic[key] = {value}
    else:
        dic[key].add(value)
    if value not in dic:
        dic[value] = {key}
    else:
        dic[value].add(key)

# 1과 연결되어있는게 하나도 없을 경우
if 1 not in dic:
    print(0)
else:
    print(len(bfs(1, dic)) - 1)  # 1 자신 제외

File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;

/*
N : 정점의 개수
M : 간선의 개수
1 : 시작 노드
*/

int N, M;
bool visited[101];
vector<int> graph[101];
int cnt = 0;

void bfs(int start_node){
    queue<int> q;
    q.push(start_node);
    visited[start_node]=true;
    
    while (!q.empty()){
        int cur_node = q.front();
        q.pop();
        for (int next_node : graph[cur_node]){
            if (visited[next_node]) continue;
            cnt++;
            visited[next_node] = true;
            q.push(next_node);
        }
    }
}

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    
    cin >> N >> M;
    for (int i=0; i<M; i++){
        int a, b;
        cin >> a >> b;
        // 무방향 그래프
        graph[a].push_back(b);
        graph[b].push_back(a);
    }

    bfs(1);
    cout << cnt;
    
    return 0;
}